Ghidra/Features/PDB/src/pdb/cpp/print.cpp

```diff
@@ -152,12 +152,16 @@ std::wstring printType( IDiaSymbol * pType, const std::wstring& suffix ) {
 		}
 		ULONGLONG lenArray = getLength( *pType );
 		ULONGLONG lenElem  = getLength( *pBaseType );
-		if (lenElem == 0) {//prevent divide by zero...
-			lenElem = lenArray;
+		ULONGLONG sz;
+		if (lenElem == 0) {
+			sz = 0;//prevent divide by zero...
+		}
+		else {
+			sz = lenArray / lenElem;
```

Comment:
The minimum length of a datatype is supposed to be 1. I believe the original intention of setting `lenElem = lenArray;` was to make `lenArray / lenElem` = 1. Of course if `lenArray` is also 0 you will get a divide by 0.

These changes are good though. The only change should be setting sz to 1 instead of 0.

Reply:
I'll do more research on this, you may be right... fwiw windbg has this to say:

```
0:000> dt ghidra_rust_pdb_bug!alloc::collections::btree::node::LeafNode<(), ()>
   +0x000 parent           : Ptr64 alloc::collections::btree::node::InternalNode<(), ()>
   +0x008 parent_idx       : core::mem::maybe_uninit::MaybeUninit<u16>
   +0x00a len              : Uint2B
   +0x00c keys             : [1] core::mem::maybe_uninit::MaybeUninit<()>
   +0x00c vals             : [1] core::mem::maybe_uninit::MaybeUninit<()>
0:000> dt ghidra_rust_pdb_bug!core::mem::maybe_uninit::MaybeUninit<()>
   +0x000 uninit           : 
   +0x000 value            : core::mem::manually_drop::ManuallyDrop<()>
```

I imagine that this is a supporting feature of zero-sized types, a rust concept for optimizing codegen around ()